### PR TITLE
Fix for FA5, update app to ember 4.12 & add tests for ember versions

### LINKED ISSUE #230: v1.0.0: 'faCheck' is not exported by node_modules/@fortawesome/free-solid-svg-icons/index.js
**Describe the bug**
Upgrade ember-fonteawesome to 1.0.0 together with `@fortawesome/free-solid-svg-icons: 5.15.4` and an icon.js config.
When running an ember 4.12.0 app,  the following error is printed:

```
'faCheck' is not exported by ../../../../../../../../Users/steve/code/myapp/node_modules/@fortawesome/free-solid-svg-icons/index.js, imported by exports-free-solid-svg-icons.js
```

Rolling back to 0.4.2 and everything works fine again. However, the changelog says that there were no changes between 0.4.2 and 1.0.0 🤷‍♀️ I have deleted my node_modules folder, to make sure this is not a local issue, but the issue persists repeatedly.

yarn version: 1.22.19

**Expected behavior**
The app still boots up after upgrading to v1.0.0.

### PR DESCRIPTION
**Pull request by mkszepp**
- fix #230 (support for FA5)
- add tests for FA5
- update to ember 4.12

### CODE REVIEW

**Comment by mkszepp** [comment]
@robmadole i have readded some code from 0.4.2 into the v1.x (which was removed). With this change, we can use also FA5 with v1.0. For FA6 there is no difference after this changes

In additional i have added a test for FA5, so we don't run into any bug with FA5.

Hopfully we can get this fix ❤️

File: package.json [hunk 1]
```diff
@@ -68,64 +70,73 @@
...
   },
   "engines": {
-    "node": "12.* || 14.* || >= 16"
+    "node": "14.* || 16.* || >= 18"
   },
   "ember": {
     "edition": "octane"
   },
+  "overrides": {
+    "ember-source": "$ember-source"
```
Comment (jrjohnson) [inline package.json:138]: Something went wrong here I think.

**Review by jrjohnson — CHANGES_REQUESTED** [review]
Thanks for doing this update! Can you pull the bugfix and test changes out of this PR and fix the issue in package.json, then I can merge this and you can re-apply your bugfix code. I'd like to keep the changes in two groups to make it easier to see any issues with either.

File: package.json (hunk 1 above)
Comment (mkszepp) [reply package.json:138]: @jrjohnson no, this is necessary to fix the peerDependency check for release, beta... tests.

The version of ember-source is in release like `ember-source@5.2.0-release+a1cd26c4`. Npm has problems in this situation. I was already running in the same problem in other projects, which were using `ember-render-modifiers` (https://github.com/emberjs/ember-render-modifiers/issues/64#issuecomment-1114913950 / https://github.com/npm/cli/issues/3932)

I know, its not the best solution, but it helps us to fix the tests.

`$ember-source` takes the version, which was specified in `devDependency`

**Comment by mkszepp** [comment]
@jrjohnson i have created to PR #234 (update ember) & #235 (fix for FA5, which has also the changes of ember update)

**Pull request closed without merging**